plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2035,6 +2147,20 @@ producer_plugin_impl::push_transaction( const fc::time_point& block_deadline,
       unset_db_read_only_mode.cancel();
    }
 
+   return handle_push_result(trx, next, start, chain, trace, return_failure_trace, disable_subjective_enforcement, first_auth, sub_bill, prev_billed_cpu_time_us);
+}
+
+producer_plugin_impl::push_result
+producer_plugin_impl::handle_push_result( const transaction_metadata_ptr& trx,
+                                          next_function<transaction_trace_ptr> next,
+                                          const fc::time_point& start,
+                                          const chain::controller& chain,
+                                          const transaction_trace_ptr& trace,
+                                          bool return_failure_trace,
+                                          bool disable_subjective_enforcement,
+                                          account_name first_auth,
+                                          int64_t sub_bill,
+                                          uint32_t prev_billed_cpu_time_us) {
```

Comment:
My core dump looks like it was caused by multiple threads writing to subjective_billing. `disable_subjective_enforcement` should be `true` for read-only.

Reply:
Will also need to not call `_account_fails` for read-only.

Reply:
Both `_time_tracker` and `_account_fails` actually.

Reply:
Sorry about this. Thank you Kevin for testing it! Fixed.